### DISCUSSION TITLE
[Feature request] Add knowledge base retrieve api into dataset open api

### DISCUSSION THREAD

**Discussion by zfanswer** [langgenius/dify#4912] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

I was trying to write my own program with Dify's knowledge base RAG function to achieve a
custom requirement, and found there is no retrieve api published in /datasets?category=api page. It was frustrating as what I can do is goning to trick the hit-testing page for single KB data retrival, also if I want to use multiple dataset, I have to combining retrival results of multiple KBs straightly w/o reranker or implementing reranker stage by my own.

### 2. Describe the feature you'd like to see

adding KB retrieve api into /datasets?category=api , at least retrieve within a single KB. 
It's even better If N-to-1 and mutiple path retrieve func can be also inclueded.

### 3. How will this feature improve your workflow or experience?

Currently, if users want to  use the RAG of knowledge base with their own programs, they can only trick the knowledge base hit-testing page for now. The knowledge base management open API lacks a retrieval API, and it would be helpful to allow users using open api with sk and enabling retrieve knowledges across multiple KBs (with multi-path reveval func).

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by zfanswer** [comment]
My issue has been converted to discussion with no comments...AGAIN

**Reply by patryk20120** [reply]
Well it is a feature request not an real issue so that is why it has been moved here.
